### PR TITLE
Background download operation skeleton

### PR DESCRIPTION
**Pull request by gennaro-safehill**
- [x] Implement `BackgroundTask` and `BackgroundTaskProcessor`
- [x] Define the interface for a `DownloadOperation`
- [x] Inject coroutine scope in background operations
- [x] Define skeleton of LocalDownloadOperation and RemoteDownloadOperation

### CODE REVIEW

File: src/main/kotlin/com/safehill/kclient/tasks/BackgroundTaskProcessor.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                    val task = taskQueue.poll()
+                    task?.run()
+                }
+                isProcessing = false
+            }
+        }
+    }
+
+    fun runRepeatedly(task: T, intervalMillis: Long) {
+        GlobalScope.launch {
```
Comment (DikenMaharjan) [inline src/main/kotlin/com/safehill/kclient/tasks/BackgroundTaskProcessor.kt]: We would need a mechanism to cancel the task as well. We would need a coroutineScope(A user bound coroutine scope or a scope whose cancellation can be triggered properly in a desired moment) to be injected into the BackgroundTaskProcessor for handling the lifecycle of the job. 
For now, lets run with GlobalScope and we can refactor it later once the working mechanism is properly implemented.

**Review by DikenMaharjan — APPROVED** [review]
Approved. Will be working on the top of these changes.

File: src/main/kotlin/com/safehill/kclient/tasks/BackgroundTaskProcessor.kt (hunk 1 above)
Comment (gennaro-safehill) [reply src/main/kotlin/com/safehill/kclient/tasks/BackgroundTaskProcessor.kt]: Both great points

**Review by FFDeveloper — APPROVED** [review]

**Review by FFDeveloper — APPROVED** [review]

**Review by DikenMaharjan — APPROVED** [review]
Approved

**Pull request merged**